### DISCUSSION TITLE
Folder names starting from capital letter

### DISCUSSION THREAD

**Discussion by lacrimal** [photoprism/photoprism#2865] [Q&A]
This is probably connected to #2672 but ... in different way :-)

I have folder ```CameraAga``` mapped to ```originals```, after indexing it is wrongly visible in ```Search```  as:
 ![obraz](https://user-images.githubusercontent.com/22749391/200198203-03821166-988d-4798-a316-dd695ad12275.png)
Trying to search using ```folder:Cameraaga``` as it is presented , gives zero results since ```folder:CameraAga``` should be used.

Now it is inconsistent across photoprism, since ```Folder``` view respect letter case:
![obraz](https://user-images.githubusercontent.com/22749391/200198305-823f5a1d-8606-4ed9-81b0-e5fe53fceb1f.png)

I think case of words should be respected across whole photoprism.

**Comment by graciousgrey** [comment]
What you see on the card view in the search is the photo title (which is autogenerated) not the folder name. You can either search for the photo by title:"Cameraga / 2022" or title:"Cameraga*" or folder:CameraAaga.

**Reply by lacrimal** [reply]
Indeed, but why letter case is not preserved since it obviously show folder named CameraAga ? It is misleading and letter case is something that is lost here.

**Reply by lastzero** [reply]
THEN AGAIN ALL MY OLD FILES AND FOLDERS FROM WINDOWS ARE IN UPPERCASE AND IM NOT SURE IF EVERYTHING FROM THAT TIME SHOULD LOOK LIKE THIS?

**Reply by lacrimal** [reply]
Not exactly sure we are talking about same thing or this is sarcastic. 
Im talking about **preserving letter case** not making everything in uppercase or lowercase or - like it is now - first capital letter then lowercase.
So if i have ```CameraAga``` it is obvious that it should be ```CameraAga``` across whole app no matter it is called ```title``` or ```folder``` from user experience perspective not understand why metadata ```title``` gallantly lose letter case of original naming. It is simply inconsistent.
If you think about this - there is reason to keep ```CameraAga``` instead ```Cameraaga``` it is simply easier to read and spot, beside it is what I did and expect. Finally ... it is not converted to ```cAMERAaGA``` or ```cAmErAaGa``` guess why ?

**Reply by lastzero** [reply]
https://superuser.com/questions/1297658/folder-names-become-uppercase-when-syncing-to-fat32-drive

So if you have "ALL WRITTEN IN CAPITAL LETTERS", it is obvious that keeping that would result in "ALL WRITTEN IN CAPITAL LETTERS".

And yes, there are edge cases like yours where it's MixeD and you want to keep that. True, that can be implemented as well. However, given the amount of other work we have, we have to ask for some patience, because other features are still completely missing. Once we have a team with specialists, for example for names and titles, they will be happy to add special rules and options.

**Reply by lacrimal** [reply]
> https://superuser.com/questions/1297658/folder-names-become-uppercase-when-syncing-to-fat32-drive
> 
> So if you have "ALL WRITTEN IN CAPITAL LETTERS", it is obvious that keeping that would result in "ALL WRITTEN IN CAPITAL LETTERS".
> 
> And yes, there are edge cases like yours where it's MixeD and you want to keep that. True, that can be implemented as well. However, given the amount of other work we have, we have to ask for some patience, because other features are still completely missing. Once we have a team with specialists, for example for names and titles, they will be happy to add special rules and options.

Don't think that this is 'edge case', more like expected behaviour. I understand that other people have problems with it so it is like it is from the beginning. I also understand that you don't have time to sort this out as it is not trivial and you have other tasks.
So this was more less reply that I expected.

**Reply by lastzero** [reply]
Mixed case, camelCase etc from file systems, some of which are not case sensitive and only supported uppercase is special. We've handled the easy cases first, which helps 95% of users and the result is that you can enjoy a stable PhotoPrism version now that has what most users need to get started. Once users provide all the funding we need, they get even more more things they need. After 5 years this is the first or second time someone comes up with this. So it can't be our top priority.

**Reply by lacrimal** [reply]
Fair enough

**Reply by lastzero** [reply]
To see all the cases we can handle well - and there are a lot - check out our test suite:
- https://github.com/photoprism/photoprism/blob/develop/pkg/txt/title_test.go#L9

If it's important to you and you want to help, feel free to send us suggestions for improvement:
- https://github.com/photoprism/photoprism/blob/develop/pkg/txt/title.go

**Reply by lastzero** [reply]
Looking at it, maybe it could already help to use this capitalization function instead of the current one at the place where the folder name is extracted, because that seems to do what you want? It still needs to be tested, so we can't do it in less than an hour or two, which we don't have right now.

**Reply by lacrimal** [reply]
I can barely understand what 'capitalization function' is :-) if it is inside [https://github.com/photoprism/photoprism/blob/develop/pkg/txt/title.go](https://github.com/photoprism/photoprism/blob/develop/pkg/txt/title.go) than probable you are right - it is what I need.
I'm not golang developer but it looks not so complicated - i will try to modify that on my side and when successful - will post it here.
